subsys/mpsl/cx/thread/mpsl_cx_thread.c

```diff
@@ -84,7 +84,7 @@ static void gpiote_irq_handler(const struct device *gpiob, struct gpio_callback
 
 	static mpsl_cx_op_map_t last_notified;
 	int32_t ret;
-	mpsl_cx_op_map_t granted_ops;
+	_Atomic mpsl_cx_op_map_t granted_ops;
```

Comment:
`_Atomic` is C11. I think we are using C99 in nRF Connect SDK

Reply:
Using `_Atomic` type does not ensure atomic memory accesses in consecutive instructions accessing this variable. Lines 101 and 102 would need to be replaced with an instruction ensuring atomic read-modify-write operation like `atomic_compare_exchange_...()`.

Reply:
And I think `last_notified` is the variable which we want to atomically read-modify-write. Not `granted_ops`